packages/@o3r/core/schematics/application/index.ts

```diff
@@ -0,0 +1,76 @@
+import { apply, chain, externalSchematic, MergeStrategy, mergeWith, move, renameTemplateFiles, Rule, Schematic, schematic, strings, template, url } from '@angular-devkit/schematics';
+import { getPackagesBaseRootFolder, getWorkspaceConfig, isNxContext } from '@o3r/schematics';
+import * as path from 'node:path';
+import type { NgGenerateApplicationSchema } from './schema';
+import type { PackageJson } from 'type-fest';
+
+/**
+ * Add an Otter application to a monorepo
+ *
+ * @param options Schematic options
+ */
+export function generateApplication(options: NgGenerateApplicationSchema): Rule {
+  const cleanName = strings.dasherize(options.name);
+
+  const addProjectSpecificFiles = (targetPath: string, rootDependencies: Record<string, string | undefined>) => {
+
+    return mergeWith(apply(url('./templates'), [
+      template({
+        ...options,
+        name: cleanName,
+        rootDependencies
+      }),
+      move(targetPath),
+      renameTemplateFiles()
+    ]), MergeStrategy.Overwrite);
+  };
+
+  const ngCliUpdate: Rule = (tree, context) => {
+
+    if (options.style) {
+      context.logger.warn('The provided "style" option will be ignored. "SCSS" will be used.');
+    }
+
+    const config = getWorkspaceConfig(tree);
+    if (!config) {
+      throw new Error('No workspace configuration file found');
+    }
```

Comment:
Why should we throw an error if no workspace config is found ?
we use it only to get the defaultRoot if no path is provided
```suggestion
    if (!options.path && !config) {
      throw new Error('The `path` option is not provided and no workspace configuration file found to define it');
    }
```